### ISSUE TITLE
subprocess.popen doesn't preserve environment

### ISSUE THREAD

**Issue by rhunwicks** [safarijv/yet-another-django-profiler#16]
Currently `ProfilerMiddleware` uses `subprocess.Popen` to run `gprof2dot.py` which in turn imports `.conf.LazySettings`, which then imports `django.conf.settings`

I am running Django using a separate settings module:

```
./manage.py runserver --settings=app.settings.profile
```

When I do this, then `PYTHONPATH` is not set in the subprocess and consequently it cannot find the settings module because `gprof2dot.py` is running in a different working directory and it fails:

```
ImportError: Could not import settings 'app.settings.profile' (Is it on sys.path? Is there an import error in the settings file?): No module named app.settings.profile
```

I can solve it by setting the `PYTHONPATH` before I start the server:

```
PYTHONPATH=./ ./manage.py runserver --settings=app.settings.profile
```

It might be worth updating the documentation to mention this.

You can also solve it by setting the `PYTHONPATH` explicitly in the subprocess:

```python
import sys
```

 to the imports, and changing the call to be:

```python
                    cmd = ('gprof2dot.py -f pstats {} | dot -Tpdf'.format(stats.name))
                    # Get a copy of the existing environment.
                    env = os.environ.copy()
                    # Add the PYTHONPATH using the existing python system path.
                    env['PYTHONPATH'] = ":".join(sys.path)
                    process = subprocess.Popen(cmd, shell=True, stdin=subprocess.PIPE,
                                               stdout=subprocess.PIPE, env=env)
```

**Comment by jmbowman** [comment]
A fix for this has been committed to master, I'll cut a new release after reviewing the remaining 2 open issues and making fixes for those also if they look simple enough.

**Comment by jmbowman** [comment]
Should be fixed in version 1.0.3
